src/pages/Dashboard/index.tsx

```diff
@@ -16,15 +19,17 @@ export default () => {
     ) {
       navigate("../login", { replace: true });
     }
-  }, []);
+  }, [navigate]);
 
   return (
     <div className="home-container flex flex-col">
-      Hello World
-      <CustomButton
-        onClick={addItem}
-        text="Add Item"
+      <CustomModal
+        open={addItemModal}
+        setOpen={setAddItemModal}
+        title={t("add_item.label")}
+        content={<AddItem setOpen={setAddItemModal} />}

```

Comment:
Please, don't use setState as prop
https://stackoverflow.com/questions/50215364/can-we-pass-setstate-as-props-from-one-component-to-other-and-change-parent-stat
Because it looks like you change state of your parent from child